src/audio/dai.c

```diff
@@ -564,6 +565,9 @@ static int dai_params(struct comp_dev *dev,
 static int dai_config_prepare(struct comp_dev *dev)
 {
 	struct dai_data *dd = comp_get_drvdata(dev);
+#if CONFIG_IMX8ULP
+	struct ipc_config_dai *dai = &dd->ipc_config;
+#endif
 	int channel = 0;
```

Comment:
We need to find another way. This is DAI generic code we don't want to pollute it with 8ulp specific code. We need to think to another solution. Allow me few days to think about it, if you don't have already some patches.